lua/entities/gmod_tardis/modules/sh_health.lua

```diff
@@ -1,228 +1,116 @@
 -- Health
 
-if SERVER then
-    ENT:AddHook("Initialize","health-init",function(self)
-        self:SetData("health-val", TARDIS:GetSetting("health-max"), true)
-        if WireLib then
-            self:TriggerWireOutput("Health", self:GetHealth())
-        end
-    end)
-
-    ENT:AddHook("SettingChanged","maxhealth-changed", function(self, id, val)
-        if id ~= "health-max" then return end
-
-        if self:GetHealth() > val then
-            self:ChangeHealth(val)
-        end
-    end)
-
-    function ENT:ChangeHealth(new_health)
-        if self:GetRepairing() then
-            return
-        end
-        local max_health = TARDIS:GetSetting("health-max")
-        if not TARDIS:GetSetting("health-enabled") then
-            self:SetData("health-val", max_health, true)
-            return
-        end
-        local old_health = self:GetHealth()
-
-        new_health = math.Clamp(new_health, 0, max_health)
-
-        self:SetData("health-val", new_health, true)
-        self:CallCommonHook("OnHealthChange", new_health, old_health)
-        if new_health == 0 and old_health ~= 0 then
-            self:CallCommonHook("OnHealthDepleted")
-        end
-    end
+function ENT:GetHealthMax()
+    local ratio = TARDIS:GetSetting("health_to_shields_ratio")
+    return math.max(1, ratio * TARDIS:GetSetting("health_max"))
 end
 
 function ENT:GetHealth()
     return self:GetData("health-val", 0)
 end
 
+function ENT:GetHealthPercent()
+    local val = self:GetHealth()
+    local percent = (val * 100) / self:GetHealthMax()
+    return percent
+end
+
+ENT.HEALTH_PERCENT_DAMAGED = 40
+ENT.HEALTH_PERCENT_BROKEN = 10
+
 function ENT:IsAlive()
     return (self:GetHealth() ~= 0)
 end
 
-function ENT:GetRepairPrimed()
-    return self:GetData("repair-primed",false)
+function ENT:IsDamaged()
+    return not self:IsBroken() and (self:GetHealthPercent() <= self.HEALTH_PERCENT_DAMAGED)
 end
 
-function ENT:GetRepairing()
-    return self:GetData("repairing",false)
+function ENT:IsBroken()
+    return (self:GetHealthPercent() <= self.HEALTH_PERCENT_BROKEN)
 end
 
-function ENT:GetHealthPercent()
-    local val = self:GetData("health-val", 0)
-    local percent = (val * 100)/TARDIS:GetSetting("health-max")
-    return percent
+function ENT:HasLowHealth()
+    return self:IsBroken() or self:IsDamaged()
 end
 
-function ENT:GetRepairTime()
-    return self:GetData("repair-time")-CurTime()
+function ENT:IsDead()
+    return (self:GetHealth() == 0)
```

Comment:
Nitpick: Brackets not required

```suggestion
    return self:GetHealth() == 0
```

Reply:
569f79c45eb4cde29cecd27cf337a25eba5f7f4d